x/hub-genesis/keeper/msg_server.go

```diff
@@ -0,0 +1,85 @@
+package keeper
+
+import (
+	"context"
+	"encoding/json"
+
+	errorsmod "cosmossdk.io/errors"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	transfertypes "github.com/cosmos/ibc-go/v6/modules/apps/transfer/types"
+	clienttypes "github.com/cosmos/ibc-go/v6/modules/core/02-client/types"
+	channeltypes "github.com/cosmos/ibc-go/v6/modules/core/04-channel/types"
+	"github.com/dymensionxyz/dymension-rdk/x/hub-genesis/types"
+	"github.com/dymensionxyz/gerr-cosmos/gerrc"
+)
+
+const port = transfertypes.PortID
+
+var _ types.MsgServer = msgServer{}
+
+type msgServer struct{ Keeper }
+
+func NewMsgServerImpl(keeper Keeper) types.MsgServer {
+	return &msgServer{Keeper: keeper}
+}
+
+// note that this is only allowed by the whitelisted relayer (enforced in ante)
+func (m msgServer) SendTransfer(goCtx context.Context, msg *types.MsgSendTransfer) (*types.MsgSendTransferResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+	err := m.SendGenesisTransfer(ctx, msg.ChannelId)
+	if err != nil {
+		return nil, err
+	}
+	return &types.MsgSendTransferResponse{}, nil
+}
+
+func (k Keeper) SendGenesisTransfer(ctx sdk.Context, channelID string) error {
+	state := k.GetState(ctx)
+	if state.InFlight {
+		return gerrc.ErrFailedPrecondition.Wrap("sent transfer is already in flight")
+	}
+	c, ok := k.channelKeeper.GetChannel(ctx, port, channelID)
+	if !ok {
+		return gerrc.ErrNotFound.Wrap("channel")
+	}
+	if c.State != channeltypes.OPEN {
+		return gerrc.ErrFailedPrecondition.Wrap("channel not open")
+	}
+	if !state.CanonicalHubTransferChannelHasBeenSet() {
+		state.SetCanonicalTransferChannel(port, channelID)
+	}
+	if state.HubPortAndChannel.Channel != channelID {
```

Comment:
lets assume a bug in relayer and sent wrong channel id with the hub.
in that case after first attempt we're gonna set it (`SetCanonicalTransferChannel`).
Than we find out it's acutally a different channel which is the canonical on the hub.
Than we try again, but fail with `channel id mismatch`. 

Why dot we not simply try every time with the passed channel given we're not in flight (vs setting the first channel attempt as the only one)? ( we will need to clear the canonical transfser channel on ack err though)